src/action/common/configure_enterprise_edition_init_service.rs

```diff
@@ -0,0 +1,188 @@
+use std::path::PathBuf;
+
+#[cfg(target_os = "macos")]
+use serde::{Deserialize, Serialize};
+#[cfg(target_os = "macos")]
+use tokio::io::AsyncWriteExt;
+use tokio::process::Command;
+use tracing::{span, Span};
+
+use crate::action::{ActionError, ActionErrorKind, ActionTag, StatefulAction};
+use crate::execute_command;
+
+use crate::action::{Action, ActionDescription};
+
+#[cfg(target_os = "macos")]
+const DARWIN_ENTERPRISE_EDITION_DAEMON_DEST: &str =
+    "/Library/LaunchDaemons/systems.determinate.nix-daemon.plist";
+/**
+Configure the init to run the Nix daemon
+*/
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
+pub struct ConfigureEnterpriseEditionInitService {
+    start_daemon: bool,
+}
+
+impl ConfigureEnterpriseEditionInitService {
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn plan(start_daemon: bool) -> Result<StatefulAction<Self>, ActionError> {
+        Ok(Self { start_daemon }.into())
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "configure_enterprise_edition_init_service")]
+impl Action for ConfigureEnterpriseEditionInitService {
+    fn action_tag() -> ActionTag {
+        ActionTag("configure_enterprise_edition_init_service")
+    }
+    fn tracing_synopsis(&self) -> String {
+        "Configure the Determinate Nix Enterprise Edition daemon related settings with launchctl"
+            .to_string()
+    }
+
+    fn tracing_span(&self) -> Span {
+        span!(
+            tracing::Level::DEBUG,
+            "configure_enterprise_edition_init_service"
+        )
+    }
+
+    fn execute_description(&self) -> Vec<ActionDescription> {
+        let mut explanation = vec![format!("Create `{DARWIN_ENTERPRISE_EDITION_DAEMON_DEST}`")];
+        if self.start_daemon {
+            explanation.push(format!(
+                "Run `launchctl load {DARWIN_ENTERPRISE_EDITION_DAEMON_DEST}`"
+            ));
+        }
+
+        vec![ActionDescription::new(self.tracing_synopsis(), explanation)]
+    }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn execute(&mut self) -> Result<(), ActionError> {
```

Comment:
It _would_ be nice to share more code with `configure_init_service`, but we can do that in a follow-up PR.